pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1074,7 +1074,9 @@ func (dr *DashboardServiceImpl) GetDashboardUIDByID(ctx context.Context, query *
 			return nil, err
 		}
 
-		if len(result) != 1 {
+		if len(result) == 0 {
+			return nil, dashboards.ErrDashboardNotFound
```

Comment:
This is needed for this to work with unified storage - otherwise we fail on getting connections